sds_data_manager/constructs/processing_construct.py

```diff
@@ -0,0 +1,103 @@
+"""Processing construct for job scheduling/execution.
+
+We will use batch processing to schedule and execute jobs. This construct
+will create the necessary resources for batch processing.
+"""
+
+from typing import Optional
+
+import aws_cdk as cdk
+from aws_cdk import aws_batch as batch
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_ecr as ecr
+from aws_cdk import aws_ecs as ecs
+from constructs import Construct
+
+
+class ProcessingConstruct(Construct):
+    """Construct for processing jobs."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        vpc: ec2.Vpc,
+        volumes: Optional[list] = None,
+        **kwargs,
+    ):
+        """Set up the primary processing environment and queue.
+
+        Additional job definitions can be added to the construct later.
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        vpc : ec2.Vpc
+            VPC into which to launch the compute instance.
+        volumes : list, optional
+            List of volumes to attach to the compute instance, by default None.
+        kwargs : dict
+            Keyword arguments.
+        """
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Create compute environment
+        compute_environment = batch.FargateComputeEnvironment(
+            self,
+            "ProcessingComputeEnvironment-spot",
+            vpc=vpc,
+            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PUBLIC),
+            spot=True,
+        )
+
+        # Create job queue
+        self.job_queue = batch.JobQueue(
+            self,
+            "ProcessingJobQueue",
+            compute_environments=[
+                batch.OrderedComputeEnvironment(
+                    compute_environment=compute_environment, order=1
+                )
+            ],
+        )
+
+        self.volumes = volumes
+
+    def add_job(self, job_name: str):
+        """Create an ECR repo and a job definition for the given job.
+
+        Parameters
+        ----------
+        job_name : str
+            Name of the job for which to create the job definition.
+        """
+        # Create a registry for each job definition (swe-repo)
+        container_repo = ecr.Repository(
```

Comment:
Oh I like how you solve the instrument/level vs only instrument ecr problem here.